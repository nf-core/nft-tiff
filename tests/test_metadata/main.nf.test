nextflow_process {

    name "Validate tiff metadata"
    script "./main.nf"
    process "COPY_TIFF"

    tag "methods"

    test("check can read tiff") {
        when {
            process {
                """
                input[0] = Channel.of(
                    file("${baseDir}/tests/mindagap.mouse_heart.wga.tiff", checkIfExists:true),
                )
                """
            }
        }
        then {
            def tiff = path(process.out.tiff[0]).tiff
            assert tiff.meta != null
            assert tiff.bitmaps != null
        }
    }

    test("check compare same tiff") {
        when {
            process {
                """
                input[0] = Channel.of(
                    file("${baseDir}/tests/mindagap.mouse_heart.wga.tiff", checkIfExists:true),
                )
                """
            }
        }
        then {
            def tiff = path(process.out.tiff[0]).tiff
            def reference = path("${baseDir}/tests/mindagap.mouse_heart.wga.tiff").tiff
            assert tiff.meta == reference.meta
            assert tiff.bitmaps == reference.bitmaps
        }
    }

    test("detect tiff pixel difference") {
        when {
            process {
                """
                input[0] = Channel.of(
                    file("${baseDir}/tests/nuclear_image.tif", checkIfExists:true),
                )
                """
            }
        }
        then {
            def tiff = path(process.out.tiff[0]).tiff
            def reference = path("${baseDir}/tests/nuclear_image_modified.tif").tiff
            assert tiff.meta == reference.meta

            try {
                tiff.bitmaps == reference.bitmaps
                assert false : "Should have detected pixel difference between images!"
            } catch (Exception e) {
                assert e.message == "Bitmap 'pixel' does not match: '138' != '160'"
            }
        }
    }

}
