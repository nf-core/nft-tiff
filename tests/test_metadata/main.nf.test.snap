{
    "check bitmap equivalence with checksums": {
        "content": [
            "8049CA66E476C46D940661BF56FE73A5"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-26T08:39:02.544076475"
    }
}